plasmapy/diagnostics/langmuir.py

```diff
@@ -1,153 +1,1073 @@
+
+    return_arg : bool, optional
+        Controls whether or not the argument of the floating potential within
+        the characteristic array should be returned instead of the value of the
+        voltage. Default is False.
+
+    Returns
+    -------
+    V_F : Quantity
+        Estimate of the floating potential in units convertible to V.
 
     Notes
     -----
-    Due to the use of different methods to obtain `n_e` and `n_i`, the ratio
-    between these does not necessarily give an accurate average ionization.
+    The method used in this function takes the probe current closest to zero
+    Amperes as the floating potential.
+
     """
-    raise NotImplementedError
 
+    probe_characteristic.check_validity()
+
+    arg_V_F = np.argmin(np.abs(probe_characteristic.current))
+
+    if(return_arg):
+        return probe_characteristic.bias[arg_V_F], arg_V_F
+
+    return probe_characteristic.bias[arg_V_F]
 
-def obtain_EEDF(potential_sweep, measured_current, probe):
-    r"""
-    Obtain the Electron Energy Distribution Function of the plasma
-    from the second derivative of the measured current response using
-    Druyvestein's equations.
+
+def get_electron_saturation_current(probe_characteristic):
+    r"""Obtain an estimate of the electron saturation current corresponding
+    to the obtained plasma potential.
 
     Parameters
     ----------
-    potential_sweep : ~astropy.units.Quantity
-        The swept electric potential applied to the probe in units
-        convertible to V.
+    probe_characteristic : Characteristic
+        The probe characteristic that is being analyzed.
+
+    Returns
+    -------
+    I_e : Quantity
+        Estimate of the electron saturation current in units convertible to A.
+
+    Notes
+    -----
+    The function `get_plasma_potential` is used to obtain an estimate of the
+    plasma potential. The corresponding electron saturation current is
+    returned.
+
+    """
+
+    probe_characteristic.check_validity()
 
-    measured_current : ~astropy.units.Quantity
-        The corresponding measured current in units convertible to A.
+    _, arg_V_P = get_plasma_potential(probe_characteristic, return_arg=True)
+
+    return probe_characteristic.current[arg_V_P]
+
+
+def get_ion_saturation_current(probe_characteristic):
+    r"""Implement the simplest but crudest method for obtaining an estimate of
+    the ion saturation current from the probe characteristic.
+
+    Parameters
+    ----------
+    probe_characteristic : Characteristic
+        The probe characteristic that is being analyzed.
+
+    Returns
+    -------
+    I_i : Quantity
+        Estimate of the ion saturation current in units convertible to A.
+
+    Notes
+    -----
+    The method implemented in this function takes the ion saturation current
+    as the smallest probe current in the characteristic. This assumes the bias
+    range in the ion region is sufficiently negative for the ion current to
+    saturate.
```

Comment:
Could there be a way to automatically detect the bias range being not sufficiently negative to warn the user?